src/libs/ReportUtils.ts

```diff
@@ -929,57 +784,53 @@ function getDefaultWorkspaceAvatar(workspaceName) {
         .replace(/[^0-9a-z]/gi, '')
         .toUpperCase();
 
-    return !alphaNumeric ? defaultWorkspaceAvatars.WorkspaceBuilding : defaultWorkspaceAvatars[`Workspace${alphaNumeric[0]}`];
+    const defaultWorkspaceAvatar = defaultWorkspaceAvatars[`Workspace${alphaNumeric[0]}`] as React.FC<SvgProps>;
+
+    return !alphaNumeric ? defaultWorkspaceAvatars.WorkspaceBuilding : defaultWorkspaceAvatar;
 }
 
-function getWorkspaceAvatar(report) {
-    const workspaceName = getPolicyName(report, allPolicies);
-    return lodashGet(allPolicies, [`${ONYXKEYS.COLLECTION.POLICY}${report.policyID}`, 'avatar']) || getDefaultWorkspaceAvatar(workspaceName);
+function getWorkspaceAvatar(report: OnyxEntry<Report>) {
+    const workspaceName = getPolicyName(report, false, allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${report?.policyID}`]);
+    return allPolicies?.[`policy${report?.policyID}`]?.avatar ?? getDefaultWorkspaceAvatar(workspaceName);
 }
 
 /**
  * Returns the appropriate icons for the given chat report using the stored personalDetails.
  * The Avatar sources can be URLs or Icon components according to the chat type.
- *
- * @param {Array} participants
- * @param {Object} personalDetails
- * @returns {Array<*>}
  */
-function getIconsForParticipants(participants, personalDetails) {
-    const participantDetails = [];
+function getIconsForParticipants(participants: number[], personalDetails: OnyxCollection<PersonalDetails>) {
+    const participantDetails: Array<[number, string, string | React.FC<SvgProps>, React.FC<SvgProps> | string]> = [];
     const participantsList = participants || [];
 
-    for (let i = 0; i < participantsList.length; i++) {
-        const accountID = participantsList[i];
-        const avatarSource = UserUtils.getAvatar(lodashGet(personalDetails, [accountID, 'avatar'], ''), accountID);
-        const displayNameLogin = lodashGet(personalDetails, [accountID, 'displayName']) || lodashGet(personalDetails, [accountID, 'login'], '');
-        participantDetails.push([accountID, displayNameLogin, avatarSource, lodashGet(personalDetails, [accountID, 'fallBackIcon'])]);
+    for (const accountID of participantsList) {
+        const avatarSource = UserUtils.getAvatar(personalDetails?.[accountID]?.avatar ?? '', accountID);
+        const displayNameLogin = personalDetails?.[accountID]?.displayName ?? personalDetails?.[accountID]?.login ?? '';
```

Comment:
@kubabutkiewicz Make sure it doesn't break old logic if `personalDetails?.[accountID]?.displayName` is `''`